drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter.java

```diff
@@ -0,0 +1,1540 @@
+                deleteColumn( (MetadataCol52) columnToDelete );
+            } else if ( columnToDelete instanceof ConditionCol52 ) {
+                deleteColumn( (ConditionCol52) columnToDelete );
+            } else if ( columnToDelete instanceof ActionCol52 ) {
+                deleteColumn( (ActionCol52) columnToDelete );
+            }
+        }
+    }
+
+    private Set<Integer> getSelectedColumnIndexes() {
+        final Set<Integer> columnUsage = new HashSet<>();
+        for ( IGridData.SelectedCell sc : getView().getModel().getSelectedCells() ) {
+            columnUsage.add( sc.getColumnIndex() );
+        }
+        return columnUsage;
+    }
+
+    private int findUiColumnIndex( final int modelColumnIndex ) {
+        final List<IGridColumn<?>> columns = uiModel.getColumns();
+        for ( int uiColumnIndex = 0; uiColumnIndex < columns.size(); uiColumnIndex++ ) {
+            final IGridColumn<?> c = columns.get( uiColumnIndex );
+            if ( c.getIndex() == modelColumnIndex ) {
+                return uiColumnIndex;
+            }
+        }
+        throw new IllegalStateException( "Column was not found!" );
+    }
+
+    @Override
+    public void onDeleteSelectedRows() {
+        assertNotReadOnly();
+        Set<Integer> selectedRowIndexes;
+        while ( !( selectedRowIndexes = getSelectedRowIndexes() ).isEmpty() ) {
+            final int rowIndex = selectedRowIndexes.iterator().next();
+            deleteRow( rowIndex );
+        }
+    }
+
+    private void deleteRow( final int rowIndex ) {
+        try {
+            synchronizer.deleteRow( rowIndex );
+            view.getLayer().draw();
+
+            //Log deletion of column
+            model.getAuditLog().add( new DeleteRowAuditLogEntry( identity.getIdentifier(),
+                                                                 rowIndex ) );
+        } catch ( ModelSynchronizer.MoveColumnVetoException e ) {
+            //Swallow
+        }
+    }
+
+    @Override
+    public void setMerged( final boolean merged ) {
+        uiModel.setMerged( merged );
+        view.getLayer().draw();
+    }
+
+    @Override
+    public boolean isMerged() {
+        return uiModel.isMerged();
+    }
+
+    @Override
+    public void showAuditLog() {
+        auditLog.show();
+    }
+
+    @Override
+    public void onInsertRowAbove() {
+        assertNotReadOnly();
+        final Set<Integer> selectedRowIndexes = getSelectedRowIndexes();
+        if ( selectedRowIndexes.size() != 1 ) {
+            return;
+        }
+        final int rowIndex = selectedRowIndexes.iterator().next();
+        insertRow( rowIndex );
+    }
+
+    @Override
+    public void onInsertRowBelow() {
+        assertNotReadOnly();
+        final Set<Integer> selectedRowIndexes = getSelectedRowIndexes();
+        if ( selectedRowIndexes.size() != 1 ) {
+            return;
+        }
+        final int rowIndex = selectedRowIndexes.iterator().next();
+        insertRow( rowIndex + 1 );
+    }
```

Comment:
The `onInsertRow*(..)` methods differ only in the index on the last line.. Might be better to extract the common code.